omniledger/contracts/coins_test.go

```diff
@@ -22,27 +23,29 @@ func TestCoin_Spawn(t *testing.T) {
 	// Testing spawning of a new coin and checking it has zero coins in it.
 	ct := cvTest{}
 	inst := omniledger.Instruction{
-		InstanceID: omniledger.NewInstanceID(nil),
 		Spawn: &omniledger.Spawn{
 			ContractID: ContractCoinID,
 		},
 	}
 
+	zero := [32]byte{}
+	zerod := darc.ID(zero[:])
+
 	c := []omniledger.Coin{}
 	sc, co, err := ContractCoin(ct, inst, c)
 	require.Nil(t, err)
 	require.Equal(t, 1, len(sc))
-	ca := omniledger.InstanceID{DarcID: make([]byte, 32), SubID: omniledger.NewSubID(inst.Hash())}
+	ca := omniledger.InstanceIDFromSlice(inst.Hash())
```

Comment:
If you decide change to use `inst.DeriveID` in the `CoinContract`, then you'll also need to change it here.